gc/verbose/VerboseHandlerOutput.cpp

```diff
@@ -929,10 +929,22 @@ MM_VerboseHandlerOutput::handleConcurrentStart(J9HookInterface** hook, UDATA eve
 void
 MM_VerboseHandlerOutput::handleConcurrentEnd(J9HookInterface** hook, UDATA eventNum, void* eventData)
 {
-	MM_ConcurrentGlobalPhaseEndEvent *event = (MM_ConcurrentGlobalPhaseEndEvent *)eventData;
-	MM_ConcurrentPhaseStatsBase *stats = (MM_ConcurrentPhaseStatsBase *)event->concurrentStats;
+	MM_ConcurrentPhaseStatsBase *stats = NULL;
+	MM_EnvironmentBase *env = NULL;
+
+	if (J9HOOK_MM_PRIVATE_CONCURRENT_GLOBAL_PHASE_END == eventNum) {
+		MM_ConcurrentGlobalPhaseEndEvent *event = (MM_ConcurrentGlobalPhaseEndEvent *)eventData;
+		stats = (MM_ConcurrentPhaseStatsBase *)event->concurrentStats;
+		env = MM_EnvironmentBase::getEnvironment(event->currentThread);
+	} else if (J9HOOK_MM_PRIVATE_CONCURRENT_LOCAL_PHASE_END == eventNum) {
+		MM_ConcurrentLocalPhaseEndEvent *event = (MM_ConcurrentLocalPhaseEndEvent *)eventData;
+		stats = (MM_ConcurrentPhaseStatsBase *)event->concurrentStats;
+		env = MM_EnvironmentBase::getEnvironment(event->currentThread);
+	} else {
+		Assert_MM_unreachable();
+	}
+
 	MM_VerboseWriterChain* writer = _manager->getWriterChain();
```

Comment:
Could you please move these other variable declarations/inits to top of method with env & stats